source/core/src/main/com/csse3200/game/components/items/ItemActions.java

```diff
@@ -324,10 +327,13 @@ private boolean fertilise(TerrainTile tile) {
    */
   private boolean plant(TerrainTile tile) {
     if (isCropTile(tile.getOccupant())) {
+      CropTileComponent cropTileComponent = tile.getOccupant().getComponent(CropTileComponent.class);
+      if (cropTileComponent != null && cropTileComponent.getPlant() == null) {
+        // Planting using seeds should remove the item from player inventory
```

Comment:
same here, also would want to move the other code inside the if so you get correct bool return and you dont trigger an event that will do nothing

Reply:
Same here

Reply:
I can't move it to the other side cause if it is a croptilecomponent and the plant/fertilize method is called, cause then getPlant will always return an entity. We need to check that before we plant something on the tile or fertilise the tile that the tile does not have a plant or the tile is not fertilised

Reply:
I dont understand, just bring up triggering the plant event after it checks that there is no plant and not regardless of a plant being there or not?

Reply:
ye sorry I misunderstood what u meant but I've updated the comment and code now